parsl/executors/status_handling.py

```diff
@@ -22,22 +22,22 @@ def __init__(self, provider):
         self._simulated_status = {}
         self._executor_bad_state = threading.Event()
         self._executor_exception = None
-        self._generated_job_id_counter = 1
+        self._generated_block_id_counter = 1
         self._tasks = {}  # type: Dict[object, Future]
 
-    def _make_status_dict(self, job_ids: List[Any], status_list: List[JobStatus]) -> Dict[Any, JobStatus]:
-        """Given a list of job ids and a list of corresponding status strings,
+    def _make_status_dict(self, block_ids: List[Any], status_list: List[JobStatus]) -> Dict[Any, JobStatus]:
+        """Given a list of block ids and a list of corresponding status strings,
         returns a dictionary mapping each job id to the corresponding status
 
```

Comment:
Does this return a dictionary with job id or block id keys?

Reply:
with block_id as keys. This is what this PR is supposed to fix, make all the keys consistent since `block_id` should never be None, while `job_id` could be

Reply:
ok - the docstring is wrong - it still talks about job IDs

Reply:
both of the `Any` types should turn into str too

Reply:
Yea, I noticed them and fixed.

Reply:
the docstring is still wrong